cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DistributedRunRecordCorrectorService.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.services;
+
+import co.cask.cdap.app.runtime.ProgramRuntimeService;
+import co.cask.cdap.app.store.Store;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A distributed-mode only run record corrector service that corrects run records at a scheduled, configurable rate.
+ */
+public class DistributedRunRecordCorrectorService extends AbstractRunRecordCorrectorService {
+  private static final Logger LOG = LoggerFactory.getLogger(DistributedRunRecordCorrectorService.class);
+
+  private final CConfiguration cConf;
+  private final ScheduledExecutorService scheduledExecutorService;
+
+  @Inject
+  public DistributedRunRecordCorrectorService(CConfiguration cConf, Store store,
+                                              ProgramLifecycleService programLifecycleService,
+                                              ProgramRuntimeService runtimeService) {
+    super(store, programLifecycleService, runtimeService);
+    this.cConf = cConf;
+    this.scheduledExecutorService = Executors.newScheduledThreadPool(1);
```

Comment:
You should create it in `startUp` instead of in constructor to avoid thread leakage.